app/service/group_member_service.rb

```diff
@@ -0,0 +1,46 @@
+class GroupMemberService
+  Row = Struct.new(:name, :email, :role, :actions, :membership)
+
+  attr_reader :group, :current_user
+
+  class << self
+    def call(**args)
+      new(**args)
+    end
+  end
+
+  def initialize(group:, current_user:)
+    @group = group
+    @current_user = current_user
+  end
+
+  def show_actions?
+    current_user.super_admin? || current_user.is_organisations_admin?(group.organisation) || current_user.is_group_admin?(group)
+  end
+
+  def rows
```

Comment:
Is it worth storing this in some way, as we're calling it a couple of times in the view, and it'll be running through the row setup each time

Reply:
I misread this but now rows is stored.